diff --git a/gxbb_p200.dts b/gxbb_p200_xds94k.dts
index 1d39c33..9fb41a4 100644
--- a/gxbb_p200.dts
+++ b/gxbb_p200_xds94k.dts
@@ -463,7 +463,7 @@
             ocr_avail = <0x200080>;          /**VDD voltage 3.3 ~ 3.4 */
             caps = "MMC_CAP_8_BIT_DATA","MMC_CAP_MMC_HIGHSPEED",
 			"MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE","MMC_CAP_1_8V_DDR",
-			"MMC_CAP_HW_RESET", "MMC_CAP_ERASE", "MMC_CAP_CMD23";
+			"MMC_CAP_HW_RESET", "MMC_CAP_ERASE";
             caps2 = "MMC_CAP2_HS200";
 			f_min = <300000>;
             f_max = <200000000>;
@@ -492,7 +492,7 @@
 				"MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE",
 				"MMC_CAP_UHS_SDR12","MMC_CAP_UHS_SDR25",
 				"MMC_CAP_UHS_SDR50","MMC_CAP_UHS_SDR104",
-				"MMC_PM_KEEP_POWER";
+				"MMC_PM_KEEP_POWER","MMC_CAP_SDIO_IRQ";
 				f_min = <400000>;
 				f_max = <200000000>;
 				max_req_size = <0x20000>;          /**128KB*/
@@ -867,13 +867,13 @@
 		pl-periph-id = <0>; /** lm name */
 		clock-src = "usb0"; /** clock src */
 		port-id = <0>;  /** ref to mach/usb.h */
-		port-type = <0>;	/** 0: otg, 1: host, 2: slave */
+		port-type = <1>;	/** 0: otg, 1: host, 2: slave */
 		port-speed = <0>; /** 0: default, high, 1: full */
 		port-config = <0>; /** 0: default */
 		port-dma = <0>; /** 0: default, 1: single, 2: incr, 3: incr4, 4: incr8, 5: incr16, 6: disable*/
-		port-id-mode = <0>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
-		gpio-vbus-power = "GPIODV_24";
-		gpios = <&gpio       GPIODV_24       GPIO_ACTIVE_HIGH>;
+		port-id-mode = <1>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
+		gpio-vbus-power = "GPIODV_26";
+		gpios = <&gpio       GPIODV_26       GPIO_ACTIVE_HIGH>;
 		gpio-work-mask	= <1>; /**0: work on pulldown,1:work on pullup*/
 		phy-reg = <0xc0000000>;
 		phy-reg-size = <0x20>;
@@ -1018,6 +1018,7 @@
 	    irq_keydown = <7>;
 	};
 
+/*
 	adc_keypad{
 		compatible = "amlogic, adc_keypad";
 		status = "okay";
@@ -1028,6 +1029,7 @@
 		key_val = <0 143 271 393 510>; //voltage=0/252/478/692/824mV, val=voltage/1800mV*1023
 		key_tolerance = <40 40 40 40 40>;
 	};
+*/
 
 	sysled {
 		compatible = "amlogic, sysled";
@@ -1185,9 +1187,9 @@
 	};
 
 	audio_spdif_pins:audio_pin1{
-		amlogic,setmask=<AO 0x10000>;	/*spdif_out*/
-		amlogic,clrmask=<AO 0x40000>;	/*spdif_out*/
-		amlogic,pins ="GPIOAO_6";		/*spdif_out*/
+		amlogic,setmask=<0x1 0x200>;	/*spdif_out*/
+		amlogic,clrmask=<0x1 0x10000>;	/*spdif_out*/
+		amlogic,pins ="GPIOY_12";		/*spdif_out*/
 	};
 
 	audio_btpcm_pins:audio_btpcm_pins{
