diff --git a/gxbb_p200.dts b/gxbb_p200_k1_plus.dts
index 1769407..bad7707 100644
--- a/gxbb_p200.dts
+++ b/gxbb_p200_k1_plus.dts
@@ -190,15 +190,15 @@
 			status = "okay";
 	};
 	ethmac: ethernet@0xc9410000{
-                compatible = "amlogic, gxbb-rgmii-dwmac";
+                compatible = "amlogic, gxbb-rmii-dwmac";
                 reg = <0x0 0xc9410000 0x0 0x10000
                 0x0 0xc8834540 0x0 0x8>;
                 interrupts = <0 8 1>;
-                phy-mode= "rgmii";
+                phy-mode= "rmii";
                 pinctrl-names = "eth_pins";
                 pinctrl-0 = <&eth_pins>;
                 rst_pin-gpios = <&gpio GPIOZ_14 0>;
-                mc_val = <0x1621>;
+                mc_val = <0x1800>;
                 resets = <&clock GCLK_IDX_ETHERNET>;
                 reset-names = "ethpower";
                 interrupt-names = "macirq";
@@ -206,23 +206,6 @@
                 clock-names = "ethclk81";
         };
 
-	pwm {
-		compatible = "amlogic, gx-pwm";
-		status = "okay";
-		pwm-outputs = <0>,<1>,<2>,<3>,<4>,<5>,<6>,<7>;
-		reg = <0x0 0xc1108550 0x0 0x30>,
-				<0x0 0xc8100550 0x0 0x10>;
-		clocks = <&clock CLK_XTAL>,
-				<&clock CLK_VID_PLL>,
-				<&clock CLK_FPLL_DIV4>,
-				<&clock CLK_FPLL_DIV3>;
-		clock-names = "xtal",
-				"vid_pll_clk",
-				"fclk_div4",
-				"fclk_div3";
-		clock-select = <0x0>,<0x0>,<0x0>,<0x0>,<0x3>,<0x0>,<0x0>,<0x0>;
-	};
-
 	mesonstream {
 		compatible = "amlogic, codec, streambuf";
 		dev_name = "mesonstream";
@@ -371,6 +354,7 @@
 			3: 250.0M    4: 333.3M    5: 400.0M
 			6: 500.0M    7: 666.7M */
 	};
+	/*
 	bt-dev{
 		compatible = "amlogic, bt-dev";
 		dev_name = "bt-dev";
@@ -378,6 +362,7 @@
 		gpio_reset = <&gpio       GPIOX_20       GPIO_ACTIVE_HIGH>;
 		//power_low_level;
 	};
+	*/
 	bluesleep{
 		compatible = "amlogic, bluesleep";
 		dev_name = "bluesleep";
@@ -402,9 +387,9 @@
 		status = "okay";
 		interrupt_pin = <&gpio       GPIOX_7       GPIO_ACTIVE_HIGH>;
 		interrupts = <	0 68 4>;
-		irq_trigger_type = "GPIO_IRQ_LOW";
+		irq_trigger_type = "GPIO_IRQ_HIGH";
 		power_on_pin = <&gpio       GPIOX_6       GPIO_ACTIVE_HIGH>;
-		dhd_static_buf;
+		power_on_pin2 = <&gpio       GPIOX_20       GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "wifi_32k_pins";
 		pinctrl-0 = <&wifi_32k_pins>;
 	};
@@ -480,7 +465,7 @@
 		dev_name = "aml_newsd.0";
 		status = "okay";
 		reg = <0x0 0xd0070000 0x0 0x2000>;
-		interrupts = <	0 216 1>;
+		interrupts = <	0 216 4>;
 		pinctrl-names =  "sdio_clk_cmd_pins", "sdio_all_pins";
 		pinctrl-0 = <&sdio_clk_cmd_pins>;
 		pinctrl-1 = <&sdio_all_pins>;
@@ -871,9 +856,7 @@
 		port-speed = <0>; /** 0: default, high, 1: full */
 		port-config = <0>; /** 0: default */
 		port-dma = <0>; /** 0: default, 1: single, 2: incr, 3: incr4, 4: incr8, 5: incr16, 6: disable*/
-		port-id-mode = <1>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
-		gpio-vbus-power = "GPIODV_24";
-		gpios = <&gpio       GPIODV_24       GPIO_ACTIVE_HIGH>;
+		port-id-mode = <0>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
 		gpio-work-mask	= <1>; /**0: work on pulldown,1:work on pullup*/
 		phy-reg = <0xc0000000>;
 		phy-reg-size = <0x20>;
@@ -1133,10 +1116,12 @@
 		};
 	};
 
-/*
 	dvb {
-		compatible = "amlogic, dvb";
-		dev_name = "dvb";
+		compatible = "amlogic,dvb";
+		// dev_name = "amlogic-dvb";
+		status = "okay";
+		fec_reset_gpio-gpios = <&gpio GPIOY_13 GPIO_ACTIVE_HIGH>;
+		power_ctrl_gpio-gpios = <&gpio GPIOY_11 GPIO_ACTIVE_LOW>;
 		//"parallel","serial","disable"
 		ts0 = "parallel";
 		ts0_control = <0>;
@@ -1150,29 +1135,36 @@
 				&clock GCLK_IDX_HIU_PARSER_TOP>;
 		reset-names = "demux", "asyncfifo", "ahbarb0", "uparsertop";
 	};
+	
 	dvbfe {
-		compatible = "amlogic, dvbfe";
-		dev_name = "dvbfe";
-		dtv_demod0 = "Mxl241";
-		dtv_demod0_i2c_adap_id = <2>;
-		dtv_demod0_i2c_addr = <0x60>;
+		compatible = "amlogic,dvbfe";
+		//dev_name = "avl6211";
+		status = "okay";
+		dtv_demod0 = "Avl6211";
+		dtv_demod0_i2c_adap_id = <1>;
+		dtv_demod0_i2c_addr = <0xC0>;
 		dtv_demod0_reset_value = <0>;
-		dtv_demod0_reset_gpio-gpios = <&gpio GPIOY_12 GPIO_ACTIVE_HIGH>;
+		//dtv_demod0_reset_gpio = <&gpio GPIOY_13 GPIO_ACTIVE_HIGH>;
+		dtv_demod0_reset_gpio-gpios = <&gpio GPIOY_13 GPIO_ACTIVE_LOW>;
+		dtv_demod0_power_gpio-gpios = <&gpio GPIOY_11 GPIO_ACTIVE_LOW>;
 		fe0_dtv_demod = <0>;
 		fe0_ts = <0>;
 		fe0_dev = <0>;
 	};
-*/
+
 };
 &i2c_a {
-  status = "disabled";
-  /*p200: multiplex with usb PWR, disbaled*/
+  status = "okay";
 };
-/*
 &i2c_b {
   status = "okay";
 };
-*/
+&i2c_c {
+  status = "okay";
+};
+&i2c_d {
+  status = "okay";
+};
 
 &i2c_slave {
 	status = "disabled";
@@ -1185,9 +1177,9 @@
 	};
 
 	audio_spdif_pins:audio_pin1{
-		amlogic,setmask=<1 0x200>;    /*spdif_out*/
-		amlogic,clrmask=<1 0x10000>;  /*spdif_out*/
-		amlogic,pins ="GPIOY_12";     /*spdif_out*/
+		amlogic,setmask=<1 0x200>;	/*spdif_out*/
+		amlogic,clrmask=<1 0x10000>;	/*spdif_out*/
+		amlogic,pins ="GPIOY_12";	/*spdif_out*/
 	};
 
 	audio_btpcm_pins:audio_btpcm_pins{
@@ -1208,3 +1200,6 @@
 &defendkey {
 	status = "okay";
 };
+&aml_remote {
+	status = "okay";
+};
