diff --git a/gxbb_p200_2G.dts b/gxbb_p200_2G_beelink_minimxIII.dts
index a5b922b..8e48a78 100644
--- a/gxbb_p200_2G.dts
+++ b/gxbb_p200_2G_beelink_minimxIII.dts
@@ -591,7 +591,7 @@
 		system:system
 		{
 			pname = "system";
-			size = <0x0 0x40000000>;
+			size = <0x0 0x80000000>;
 			mask = <1>;
 		};
 		cache:cache
@@ -822,11 +822,11 @@
 		pl-periph-id = <0>; /** lm name */
 		clock-src = "usb0"; /** clock src */
 		port-id = <0>;  /** ref to mach/usb.h */
-		port-type = <0>;	/** 0: otg, 1: host, 2: slave */
+		port-type = <1>;	/** 0: otg, 1: host, 2: slave */
 		port-speed = <0>; /** 0: default, high, 1: full */
 		port-config = <0>; /** 0: default */
 		port-dma = <0>; /** 0: default, 1: single, 2: incr, 3: incr4, 4: incr8, 5: incr16, 6: disable*/
-		port-id-mode = <0>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
+		port-id-mode = <1>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
 		gpio-vbus-power = "GPIODV_24";
 		gpios = <&gpio       GPIODV_24       GPIO_ACTIVE_HIGH>;
 		gpio-work-mask	= <1>; /**0: work on pulldown,1:work on pullup*/
@@ -988,8 +988,8 @@
 		compatible = "amlogic, sysled";
 		dev_name = "sysled";
 		status = "okay";
-		led_gpio = <&gpio_ao  GPIOAO_13  GPIO_ACTIVE_HIGH>;
-		led_active_low = <1>;
+		led_gpio = <&gpio_ao  GPIOAO_13  GPIO_ACTIVE_LOW>;
+		led_active_low = <0>;
 	};
 	aml_sensor0: aml-sensor@0 {
 		compatible = "amlogic, aml-thermal";
