diff --git a/gxbb_p200_2G.dts b/gxbb_p200_2G_t95max.dts
index a5b922b..34940cc 100644
--- a/gxbb_p200_2G.dts
+++ b/gxbb_p200_2G_t95max.dts
@@ -631,7 +661,7 @@
 		};
 		keysn_1:key_1{
 			key-name = "mac";
-                        key-device  = "normal";
+                        key-device  = "efuse";
 			key-permit = "read","write","del";
 		};
 		keysn_2:key_2{
@@ -647,13 +677,13 @@
 		};
 		keysn_4:key_4{
 			key-name = "mac_bt";
-			key-device = "normal";
+			key-device = "efuse";
 			key-permit = "read","write","del";
                         key-type  = "mac";
 		};
 		keysn_5:key_5{
 			key-name = "mac_wifi";
-			key-device = "normal";
+			key-device = "efuse";
 			key-permit = "read","write","del";
                         key-type  = "mac";
 		};
@@ -822,11 +867,11 @@
 		pl-periph-id = <0>; /** lm name */
 		clock-src = "usb0"; /** clock src */
 		port-id = <0>;  /** ref to mach/usb.h */
-		port-type = <0>;	/** 0: otg, 1: host, 2: slave */
+		port-type = <1>;	/** 0: otg, 1: host, 2: slave */
 		port-speed = <0>; /** 0: default, high, 1: full */
 		port-config = <0>; /** 0: default */
 		port-dma = <0>; /** 0: default, 1: single, 2: incr, 3: incr4, 4: incr8, 5: incr16, 6: disable*/
-		port-id-mode = <0>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
+		port-id-mode = <1>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
 		gpio-vbus-power = "GPIODV_24";
 		gpios = <&gpio       GPIODV_24       GPIO_ACTIVE_HIGH>;
 		gpio-work-mask	= <1>; /**0: work on pulldown,1:work on pullup*/
@@ -1137,9 +1185,9 @@
 	};
 
 	audio_spdif_pins:audio_pin1{
-		amlogic,setmask=<AO 0x10000>;	/*spdif_out*/
-		amlogic,clrmask=<AO 0x40000>;	/*spdif_out*/
-		amlogic,pins ="GPIOAO_6";		/*spdif_out*/
+		amlogic,setmask=<0x1 0x200>;	/*spdif_out*/
+		amlogic,clrmask=<0x1 0x10000>;	/*spdif_out*/
+		amlogic,pins ="GPIOY_12";		/*spdif_out*/
 	};
 
 	audio_btpcm_pins:audio_btpcm_pins{
