diff --git a/gxbb_p200.dts b/gxbb_p200_mxq_pro_4k.dts
index 1d39c33..3973fb6 100644
--- a/gxbb_p200.dts
+++ b/gxbb_p200_mxq_pro_4k.dts
@@ -190,15 +190,15 @@
 			status = "okay";
 	};
 	ethmac: ethernet@0xc9410000{
-                compatible = "amlogic, gxbb-rgmii-dwmac";
+                compatible = "amlogic, gxbb-rmii-dwmac";
                 reg = <0x0 0xc9410000 0x0 0x10000
                 0x0 0xc8834540 0x0 0x8>;
                 interrupts = <0 8 1>;
-                phy-mode= "rgmii";
+                phy-mode= "rmii";
                 pinctrl-names = "eth_pins";
                 pinctrl-0 = <&eth_pins>;
                 rst_pin-gpios = <&gpio GPIOZ_14 0>;
-                mc_val = <0x1621>;
+                mc_val = <0x1800>;
                 resets = <&clock GCLK_IDX_ETHERNET>;
                 reset-names = "ethpower";
                 interrupt-names = "macirq";
@@ -480,7 +482,7 @@
 		dev_name = "aml_newsd.0";
 		status = "okay";
 		reg = <0x0 0xd0070000 0x0 0x2000>;
-		interrupts = <	0 216 1>;
+		interrupts = <	0 216 4>;
 		pinctrl-names =  "sdio_clk_cmd_pins", "sdio_all_pins";
 		pinctrl-0 = <&sdio_clk_cmd_pins>;
 		pinctrl-1 = <&sdio_all_pins>;
