diff --git a/gxbb_p200_2G.dts b/gxbb_p200_2G_iNepo_Q.dts
index a5b922b..52dfa27 100644
--- a/gxbb_p200_2G.dts
+++ b/gxbb_p200_2G_iNepo_Q.dts
@@ -631,7 +631,7 @@
 		};
 		keysn_1:key_1{
 			key-name = "mac";
-                        key-device  = "normal";
+                        key-device  = "efuse";
 			key-permit = "read","write","del";
 		};
 		keysn_2:key_2{
@@ -822,11 +822,11 @@
 		pl-periph-id = <0>; /** lm name */
 		clock-src = "usb0"; /** clock src */
 		port-id = <0>;  /** ref to mach/usb.h */
-		port-type = <0>;	/** 0: otg, 1: host, 2: slave */
+		port-type = <1>;	/** 0: otg, 1: host, 2: slave */
 		port-speed = <0>; /** 0: default, high, 1: full */
 		port-config = <0>; /** 0: default */
 		port-dma = <0>; /** 0: default, 1: single, 2: incr, 3: incr4, 4: incr8, 5: incr16, 6: disable*/
-		port-id-mode = <0>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
+		port-id-mode = <1>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
 		gpio-vbus-power = "GPIODV_24";
 		gpios = <&gpio       GPIODV_24       GPIO_ACTIVE_HIGH>;
 		gpio-work-mask	= <1>; /**0: work on pulldown,1:work on pullup*/
@@ -983,7 +983,7 @@
 		key_val = <0 143 271 393 510>; //voltage=0/252/478/692/824mV, val=voltage/1800mV*1023
 		key_tolerance = <40 40 40 40 40>;
 	};
-
+/*
 	sysled {
 		compatible = "amlogic, sysled";
 		dev_name = "sysled";
@@ -991,6 +991,7 @@
 		led_gpio = <&gpio_ao  GPIOAO_13  GPIO_ACTIVE_HIGH>;
 		led_active_low = <1>;
 	};
+*/
 	aml_sensor0: aml-sensor@0 {
 		compatible = "amlogic, aml-thermal";
 		device_name = "thermal";
@@ -1137,9 +1138,9 @@
 	};
 
 	audio_spdif_pins:audio_pin1{
-		amlogic,setmask=<AO 0x10000>;	/*spdif_out*/
-		amlogic,clrmask=<AO 0x40000>;	/*spdif_out*/
-		amlogic,pins ="GPIOAO_6";		/*spdif_out*/
+		amlogic,setmask=<0x1 0x200>;	/*spdif_out*/
+		amlogic,clrmask=<0x1 0x10000>;	/*spdif_out*/
+		amlogic,pins ="GPIOY_12";		/*spdif_out*/
 	};
 
 	audio_btpcm_pins:audio_btpcm_pins{
