diff --git a/gxl_p212_1g.dts b/gxl_p212_1g_slowemmc.dts
index 74a682b..5e6a011 100644
--- a/gxl_p212_1g.dts
+++ b/gxl_p212_1g_slowemmc.dts
@@ -448,7 +448,7 @@
 			caps = "MMC_CAP_8_BIT_DATA","MMC_CAP_MMC_HIGHSPEED",
 				 "MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE","MMC_CAP_1_8V_DDR",
 				 "MMC_CAP_HW_RESET", "MMC_CAP_ERASE", "MMC_CAP_CMD23";
-			caps2 = "MMC_CAP2_HS200", "MMC_CAP2_HS400";
+			caps2 = "MMC_CAP2_HS200";
 			f_min = <300000>;
 			f_max = <100000000>;
 			max_req_size = <0x20000>;          /**256KB*/
