diff --git a/gxl_p230_2g.dts b/gxl_p230_k1_pro.dts
index 577fc62..c69f5ed 100644
--- a/gxl_p230_2g.dts
+++ b/gxl_p230_k1_pro.dts
@@ -1219,66 +1219,49 @@
 			};
 		};
 	};
-	/* dvb {
-		compatible = "amlogic, dvb";
+	dvb {
+		compatible = "amlogic,dvb";
 		dev_name = "dvb";
-		ts1 = "serial";
-		ts1_control = <0x800>;
-		ts1_invert = <0>;
-		ts_out_invert = <0>;
-		pinctrl-names = "p_ts1", "s_ts1";
-		pinctrl-0 = <&dvb_p_ts1_pins>;
-		pinctrl-1 = <&dvb_s_ts1_pins>;
-		resets = <&clock GCLK_IDX_DEMUX
-				&clock GCLK_IDX_ASYNC_FIFO
-				&clock GCLK_IDX_AHB_ARB0
-				&clock GCLK_IDX_HIU_PARSER_TOP>;
+		ts0 = "parallel";
+		ts0_control = <0x0>;
+		ts0_invert = <0x0>;
+		fec_reset_gpio-gpios = <&gpio GPIODV_13 GPIO_ACTIVE_HIGH>;
+		power_ctrl_gpio-gpios = <&gpio GPIODV_11 GPIO_ACTIVE_LOW>;
+		pinctrl-names = "p_ts0", "s_ts0", "s_ts1";
+		pinctrl-0 = <&dvb_p_ts0_pins>;
+		pinctrl-1 = <&dvb_s_ts0_pins>;
+		pinctrl-2 = <&dvb_s_ts1_pins>;
+		resets = <&clock GCLK_IDX_DEMUX &clock GCLK_IDX_ASYNC_FIFO &clock GCLK_IDX_AHB_ARB0 &clock GCLK_IDX_HIU_PARSER_TOP>;
 		reset-names = "demux", "asyncfifo", "ahbarb0", "uparsertop";
-	}; */
-	/* dvbfe {
-		compatible = "amlogic, dvbfe";
-		dev_name = "dvbfe";
+	};
+	dvbfe {
+		compatible = "amlogic,dvbfe";
+		//dev_name = "dvbfe";
+		status = "okay";
 		dtv_demod0 = "Avl6211";
 		dtv_demod0_i2c_adap_id = <2>;
-		dtv_demod0_i2c_addr = <0xc6>;
-		dtv_demod0_reset_value = <0>;
-		dtv_demod0_reset_gpio = <&gpio GPIOZ_15 GPIO_ACTIVE_HIGH>;
+		dtv_demod0_i2c_addr = <0x60>;
+ 		dtv_demod0_reset_value = <0>;
+		dtv_demod0_reset_gpio-gpios = <&gpio GPIODV_13 GPIO_ACTIVE_LOW>;
+		dtv_demod0_power_gpio-gpios = <&gpio GPIODV_14 GPIO_ACTIVE_LOW>;
 		fe0_dtv_demod = <0>;
-		fe0_ts = <1>;
+		fe0_ts = <0>;
 		fe0_dev = <0>;
-	}; */
+	};
 
-	//dvbci {
-	//	compatible = "amlogic, dvbci";
-	//	dev_name = "dvbci";
-	//	io_type = <1>; /* 0=iobus,1=spi*/
-  //      interrupts = <	0 70 1 >;
-	//	dvbci_io {
-	//		reset_pin = <&gpio GPIODV_12 GPIO_ACTIVE_HIGH>;
-	//		cd_pin1 = <&gpio GPIODV_24 GPIO_ACTIVE_HIGH>;
-	//		cd_pin2 = <&gpio GPIODV_24 GPIO_ACTIVE_HIGH>;
-	//		pwr_pin  = <&gpio GPIODV_11 GPIO_ACTIVE_HIGH>;
-	//		irq_cam_pin = <&gpio GPIODV_28 GPIO_ACTIVE_HIGH>;
-	//		ex_tsin_en_pin = <&gpio GPIODV_25 GPIO_ACTIVE_HIGH>;
-	//		irq_cd1 = <6>;
-			/*below is spi config,*/
-		    /*if use iobus you need add iobus config below */
-	//		spi_bus_num = <0>;
-	//		spi_chip_select = <0>;
-	//		spi_max_frequency = <300000>; /* unit: hz */
-	//		spi_mode = <3>; /* mode: 0, 1, 2, 3 */
-	//		spi_cs_delay = <10 100>; /* hold_high_delay, clk_cs_delay (unit: us) */
-	//		spi_write_check = <0>; /* 0=disable, 1=enable */
-	//	};
-	//};
 };
-/*&i2c_a {
-  status = "disabled";
-};*/
-/*&i2c_b {
+&i2c_a {
   status = "okay";
-};*/
-
+};
+&i2c_b {
+  status = "okay";
+};
+&i2c_c {
+  status = "okay";
+};
+&i2c_d {
+  status = "okay";
+};
 &pinmux {
 	audio_pins:audio_pin{
 		amlogic,setmask=<6 0x7800000>;
@@ -1306,6 +1289,29 @@
 						 3 0x18>;
 		amlogic,pins = "GPIODV_24","GPIODV_25";
 	};
+
+		dvb_p_ts0_pins:dvb_p_ts0_pins {
+			amlogic,setmask = <0x2 0x1f>;
+			amlogic,clrmask = <0x3 0x787 0x2 0xff000400>;
+			amlogic,pins = "GPIODV_0", "GPIODV_1", "GPIODV_2", "GPIODV_3", "GPIODV_4", "GPIODV_5", "GPIODV_6", "GPIODV_7", "GPIODV_8", "GPIODV_9", "GPIODV_10";
+		};
+
+		dvb_s_ts0_pins:dvb_s_ts0_pins {
+			amlogic,setmask = <0x2 0x17>;
+			amlogic,clrmask = <0x3 0x584 0x2 0x7000000 0x1 0x100>;
+			amlogic,pins = "GPIODV_0", "GPIODV_8", "GPIODV_9", "GPIODV_10";
+		};
+
+		dvb_s_ts1_pins:dvb_s_ts1_pins {
+			amlogic,setmask = <0x3 0x17>;
+			amlogic,clrmask = <0x2 0xf0000 0x1 0x7>;
+			amlogic,pins = "GPIODV_0", "GPIODV_8", "GPIODV_9", "GPIODV_10";
+		};
+
+		dtv_params_pin:dtv_params_pin {
+			amlogic,clrmask = <0x1 0x38000000 0x3 0x80>;
+			amlogic,pins = "GPIODV_13", "GPIODV_14", "GPIODV_15";
+		};
 };
 &efuse {
 	status = "okay";
