diff --git a/gxl_p212_2g.dts b/gxl_p212_3g.dts
index 87fd2ee..511590f 100644
--- a/gxl_p212_2g.dts
+++ b/gxl_p212_3g.dts
@@ -27,7 +27,7 @@
 #include "mesongxl.dtsi"
 / {
 	model = "Amlogic";
-	amlogic-dt-id = "gxl_p212_2g";
+	amlogic-dt-id = "gxl_p212_3g";
 	compatible = "amlogic, Gxbb";
 	interrupt-parent = <&gic>;
 	#address-cells = <2>;
@@ -43,7 +43,7 @@
 
 	memory@00000000 {
 		device_type = "memory";
-		linux,usable-memory = <0x0 0x100000 0x0 0x7f000000>;
+		linux,usable-memory = <0x0 0x100000 0x0 0xbf000000>;
 	};
 
 	reserved-memory {
