diff --git a/gxl_p230_2g.dts b/gxl_p230_k1_pro.dts
index 0b7ce9e5..7e8f659 100644
--- a/gxl_p230_2g.dts
+++ b/gxl_p230_k1_pro.dts
@@ -543,18 +543,17 @@
 					};
 	};*/
 	partitions: partitions{
-        parts = <11>;
+        parts = <10>;
 		part-0 = <&logo>;
 		part-1 = <&recovery>;
 		part-2 = <&rsv>;
 		part-3 = <&tee>;
 		part-4 = <&crypt>;
 		part-5 = <&misc>;
-		part-6 = <&instaboot>;
-		part-7 = <&boot>;
-		part-8 = <&system>;
-		part-9 = <&cache>;
-		part-10 = <&data>;
+		part-6 = <&boot>;
+		part-7 = <&system>;
+		part-8 = <&cache>;
+		part-9 = <&data>;
 
 		logo:logo{
 			pname = "logo";
@@ -586,11 +585,6 @@
 			size = <0x0 0x2000000>;
 			mask = <1>;
 		};
-		instaboot:instaboot{
-			pname = "instaboot";
-			size = <0x0 0x20000000>;
-			mask = <1>;
-		};
 		boot:boot
 		{
 			pname = "boot";
@@ -600,7 +594,7 @@
 		system:system
 		{
 			pname = "system";
-			size = <0x0 0x40000000>;
+			size = <0x0 0x80000000>;
 			mask = <1>;
 		};
 		cache:cache
@@ -1136,8 +1130,49 @@
 			};
 		};
 	};
-};
 
+	dvb {
+		compatible = "amlogic,dvb";
+		dev_name = "dvb";
+		ts0 = "parallel";
+		ts0_control = <0x0>;
+		ts0_invert = <0x0>;
+		fec_reset_gpio-gpios = <&gpio GPIODV_13 GPIO_ACTIVE_HIGH>;
+		power_ctrl_gpio-gpios = <&gpio GPIODV_11 GPIO_ACTIVE_LOW>;
+		pinctrl-names = "p_ts0", "s_ts0", "s_ts1";
+		pinctrl-0 = <&dvb_p_ts0_pins>;
+		pinctrl-1 = <&dvb_s_ts0_pins>;
+		pinctrl-2 = <&dvb_s_ts1_pins>;
+		resets = <&clock GCLK_IDX_DEMUX &clock GCLK_IDX_ASYNC_FIFO &clock GCLK_IDX_AHB_ARB0 &clock GCLK_IDX_HIU_PARSER_TOP>;
+		reset-names = "demux", "asyncfifo", "ahbarb0", "uparsertop";
+	};
+	dvbfe {
+		compatible = "amlogic,dvbfe";
+		//dev_name = "dvbfe";
+		status = "okay";
+		dtv_demod0 = "Avl6211";
+		dtv_demod0_i2c_adap_id = <2>;
+		dtv_demod0_i2c_addr = <0x60>;
+ 		dtv_demod0_reset_value = <0>;
+		dtv_demod0_reset_gpio-gpios = <&gpio GPIODV_13 GPIO_ACTIVE_LOW>;
+		dtv_demod0_power_gpio-gpios = <&gpio GPIODV_14 GPIO_ACTIVE_LOW>;
+		fe0_dtv_demod = <0>;
+		fe0_ts = <0>;
+		fe0_dev = <0>;
+	};
+};
+&i2c_a {
+  status = "okay";
+};
+&i2c_b {
+  status = "okay";
+};
+&i2c_c {
+  status = "okay";
+};
+&i2c_d {
+  status = "okay";
+};
 &pinmux {
 	audio_pins:audio_pin{
 		amlogic,setmask=<6 0x7800000>;
@@ -1157,6 +1192,29 @@
 		amlogic,clrmask=<5 0x3c33>;
 		amlogic,pins ="GPIOX_8", "GPIOX_9", "GPIOX_10", "GPIOX_11";
 	};
+
+		dvb_p_ts0_pins:dvb_p_ts0_pins {
+			amlogic,setmask = <0x2 0x1f>;
+			amlogic,clrmask = <0x3 0x787 0x2 0xff000400>;
+			amlogic,pins = "GPIODV_0", "GPIODV_1", "GPIODV_2", "GPIODV_3", "GPIODV_4", "GPIODV_5", "GPIODV_6", "GPIODV_7", "GPIODV_8", "GPIODV_9", "GPIODV_10";
+		};
+
+		dvb_s_ts0_pins:dvb_s_ts0_pins {
+			amlogic,setmask = <0x2 0x17>;
+			amlogic,clrmask = <0x3 0x584 0x2 0x7000000 0x1 0x100>;
+			amlogic,pins = "GPIODV_0", "GPIODV_8", "GPIODV_9", "GPIODV_10";
+		};
+
+		dvb_s_ts1_pins:dvb_s_ts1_pins {
+			amlogic,setmask = <0x3 0x17>;
+			amlogic,clrmask = <0x2 0xf0000 0x1 0x7>;
+			amlogic,pins = "GPIODV_0", "GPIODV_8", "GPIODV_9", "GPIODV_10";
+		};
+
+		dtv_params_pin:dtv_params_pin {
+			amlogic,clrmask = <0x1 0x38000000 0x3 0x80>;
+			amlogic,pins = "GPIODV_13", "GPIODV_14", "GPIODV_15";
+		};
 };
 &efuse {
 	status = "okay";
