diff --git a/gxl_p212_2g.dts b/gxl_p212_2g_kvim.dts
index 87fd2ee..cda861f 100644
--- a/gxl_p212_2g.dts
+++ b/gxl_p212_2g_kvim.dts
@@ -27,7 +27,7 @@
 #include "mesongxl.dtsi"
 / {
 	model = "Amlogic";
-	amlogic-dt-id = "gxl_p212_2g";
+	amlogic-dt-id = "kvim";
 	compatible = "amlogic, Gxbb";
 	interrupt-parent = <&gic>;
 	#address-cells = <2>;
@@ -140,7 +140,7 @@
 		compatible = "amlogic, sysled";
 		dev_name = "sysled";
 		status = "okay";
-		led_gpio = <&gpio GPIODV_24 GPIO_ACTIVE_HIGH>;
+		led_gpio = <&gpio_ao GPIOAO_9 GPIO_ACTIVE_HIGH>;
 		led_active_low = <1>;
 	};
 
@@ -214,8 +224,6 @@
 			interrupts = <0 8 1
 						  0 9 1>;
 			phy-mode= "rmii";
-			pinctrl-names = "internal_eth_pins";
-			pinctrl-0 = <&internal_eth_pins>;
 			mc_val = <0x1800>;
 			resets = <&clock GCLK_IDX_ETHERNET>;
 			reset-names = "ethpower";
@@ -411,7 +411,7 @@
 		alarm_reg_addr = <0xc81000a8>;
 		timer_e_addr = <0xc1109988>;
 		init_date = "2015/01/01";
-		status = "okay";
+		status = "disabled";
 	};
 
 	wifi{
@@ -846,7 +848,7 @@
 		compatible = "amlogic, meson-uart";
 		reg = <0x0 0xc81004e0 0x0 0x18>;
 		interrupts = <0 197 1>;
-		status = "disable";
+		status = "okay";
 		clocks = <&clock CLK_XTAL>;
 		clock-names = "clk_uart";
 		fifosize = < 64 >;
@@ -907,7 +907,7 @@
 		port-id-mode = <0>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
 		usb-fifo = <728>;
 		cpu-type = "gxl";
-		controller-type = <1>; /** 0: normal, 1: otg+dwc3 host only, 2: otg+dwc3 device only*/
+		controller-type = <2>; /** 0: normal, 1: otg+dwc3 host only, 2: otg+dwc3 device only*/
 		phy-reg = <0xd0078000>;
 		phy-reg-size = <0xa0>;
 		resets = <&clock GCLK_IDX_USB_GENERAL
@@ -1020,13 +1022,11 @@
 				"Ext Spk","LOUTL",
 				"Ext Spk","LOUTR";
 
-		mute_gpio-gpios = <&gpio GPIOH_5 0>;
 		mute_inv;
 		sleep_time = <100>;
 		hp_disable;
 		hp_paraments = <800 300 0 5 1>;
 		pinctrl-names = "aml_snd_m8";
-		pinctrl-0 = <&audio_pins>;
 		cpu_list = <&cpudai0 &cpudai1 &cpudai2>;
 		codec_list = <&codec0 &codec1 &codec2>;
 		plat_list = <&i2s_plat &i2s_plat &pcm_plat>;
@@ -1091,6 +1091,17 @@
 		irq_keyup = <6>;
 		irq_keydown = <7>;
 	};
+	adc_keypad{
+		compatible = "amlogic, adc_keypad";
+		status = "okay";
+		key_name = "home";
+		key_num = <1>;
+		key_code = <300>;
+		key_chan = <0>;
+		key_val = <10>;
+		key_tolerance = <40>;
+	};
+
 	aml_sensor0: aml-sensor@0 {
 		compatible = "amlogic, aml-thermal";
 		device_name = "thermal";
@@ -1218,3 +1229,12 @@
 &defendkey {
 	status = "okay";
 };
+&i2c_b {
+	status = "okay";
+	rtc_hym8563{
+        	compatible = "amlogic, rtc_hym8563";
+	    	dev_name = "rtc_hym8563";
+	    	status = "okay";
+		reg = <0x51>;
+	};
+};
