diff --git a/gxl_p212_2g.dts b/gxl_p212_2g_kvim.dts
index 196eeeb..33d334a 100644
--- a/gxl_p212_2g.dts
+++ b/gxl_p212_2g_kvim.dts
@@ -25,7 +25,7 @@
 #include "mesongxl.dtsi"
 / {
 	model = "Amlogic";
-	amlogic-dt-id = "gxl_p212_2g";
+	amlogic-dt-id = "kvim";
 	compatible = "amlogic, Gxbb";
 	interrupt-parent = <&gic>;
 	#address-cells = <2>;
@@ -132,7 +132,7 @@
 		compatible = "amlogic, sysled";
 		dev_name = "sysled";
 		status = "okay";
-		led_gpio = <&gpio GPIODV_24 GPIO_ACTIVE_HIGH>;
+		led_gpio = <&gpio_ao GPIOAO_9 GPIO_ACTIVE_HIGH>;
 		led_active_low = <1>;
 	};
 
@@ -374,7 +374,7 @@
 		alarm_reg_addr = <0xc81000a8>;
 		timer_e_addr = <0xc1109988>;
 		init_date = "2015/01/01";
-		status = "okay";
+		status = "disabled";
 	};
 
 	wifi{
@@ -389,31 +389,6 @@
 		pinctrl-names = "wifi_32k_pins";
 		pinctrl-0 = <&wifi_32k_pins>;
 	};
-	emmc{
-		compatible = "amlogic, aml_sd_emmc";
-		dev_name = "aml_newsd.0";
-		status = "okay";
-		reg = <0x0 0xd0074000 0x0 0x2000>;
-		interrupts = <	0 218 1>;
-		pinctrl-names = "emmc_clk_cmd_pins", "emmc_all_pins";
-		pinctrl-0 = <&emmc_clk_cmd_pins>;
-		pinctrl-1 = <&emmc_conf_pull_up &emmc_conf_pull_done &emmc_all_pins>;
-		emmc{
-			status = "disabled";
-			pinname = "emmc";
-			ocr_avail = <0x200080>;          /**VDD voltage 3.3 ~ 3.4 */
-			caps = "MMC_CAP_8_BIT_DATA","MMC_CAP_MMC_HIGHSPEED",
-				 "MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE","MMC_CAP_1_8V_DDR",
-				 "MMC_CAP_HW_RESET", "MMC_CAP_ERASE", "MMC_CAP_CMD23";
-			caps2 = "MMC_CAP2_HS200", "MMC_CAP2_HS400";
-			f_min = <300000>;
-			f_max = <100000000>;
-			max_req_size = <0x20000>;          /**256KB*/
-			gpio_dat3 = <&gpio       BOOT_3      GPIO_ACTIVE_HIGH>;
-			hw_reset =  <&gpio       BOOT_9      GPIO_ACTIVE_HIGH>;
-			card_type = <1>; /* 1:mmc card(include eMMC), 2:sd card(include tSD), */
-		};
-	};
     sd{
         compatible = "amlogic, aml_sd_emmc";
         dev_name = "aml_newsd.0";
@@ -452,6 +427,31 @@
 	   		card_type = <5>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
 		};
     };
+    emmc{
+        compatible = "amlogic, aml_sd_emmc";
+        dev_name = "aml_newsd.0";
+        status = "okay";
+        reg = <0x0 0xd0074000 0x0 0x2000>;
+        interrupts = <	0 218 1>;
+        pinctrl-names = "emmc_clk_cmd_pins", "emmc_all_pins";
+        pinctrl-0 = <&emmc_clk_cmd_pins>;
+        pinctrl-1 = <&emmc_conf_pull_up &emmc_conf_pull_done &emmc_all_pins>;
+        emmc{
+            status = "disabled";
+            pinname = "emmc";
+            ocr_avail = <0x200080>;          /**VDD voltage 3.3 ~ 3.4 */
+            caps = "MMC_CAP_8_BIT_DATA","MMC_CAP_MMC_HIGHSPEED",
+					"MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE","MMC_CAP_1_8V_DDR",
+					"MMC_CAP_HW_RESET", "MMC_CAP_ERASE", "MMC_CAP_CMD23";
+            caps2 = "MMC_CAP2_HS200", "MMC_CAP2_HS400";
+			f_min = <300000>;
+            f_max = <100000000>;
+            max_req_size = <0x20000>;          /**256KB*/
+		gpio_dat3 = <&gpio       BOOT_3      GPIO_ACTIVE_HIGH>;
+		hw_reset =  <&gpio       BOOT_9      GPIO_ACTIVE_HIGH>;
+            card_type = <1>; /* 1:mmc card(include eMMC), 2:sd card(include tSD), */
+	};
+    };
 	sdio{
 		compatible = "amlogic, aml_sd_emmc";
 		dev_name = "aml_newsd.0";
@@ -863,7 +863,7 @@
 		port-id-mode = <0>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
 		usb-fifo = <728>;
 		cpu-type = "gxl";
-		controller-type = <1>; /** 0: normal, 1: otg+dwc3 host only, 2: otg+dwc3 device only*/
+		controller-type = <2>; /** 0: normal, 1: otg+dwc3 host only, 2: otg+dwc3 device only*/
 		phy-reg = <0xd0078000>;
 		phy-reg-size = <0xa0>;
 		resets = <&clock GCLK_IDX_USB_GENERAL
@@ -1018,6 +1018,17 @@
 		irq_keyup = <6>;
 		irq_keydown = <7>;
 	};
+	adc_keypad{
+		compatible = "amlogic, adc_keypad";
+		status = "okay";
+		key_name = "home";
+		key_num = <1>;
+		key_code = <300>;
+		key_chan = <0>;
+		key_val = <10>;
+		key_tolerance = <40>;
+	};
+
 	aml_sensor0: aml-sensor@0 {
 		compatible = "amlogic, aml-thermal";
 		device_name = "thermal";
@@ -1145,3 +1156,12 @@
 &defendkey {
 	status = "okay";
 };
+&i2c_b {
+	status = "okay";
+	rtc_hym8563{
+        	compatible = "amlogic, rtc_hym8563";
+	    	dev_name = "rtc_hym8563";
+	    	status = "okay";
+		reg = <0x51>;
+	};
+};
