diff --git a/gxm_q200_2g.dts b/gxm_q200_2g_minix_neo_u9.dts
index 4a46647..4a5c0a2 100644
--- a/gxm_q200_2g.dts
+++ b/gxm_q200_2g_minix_neo_u9.dts
@@ -140,7 +140,7 @@
 		compatible = "amlogic, sysled";
 		dev_name = "sysled";
 		status = "okay";
-		led_gpio = <&gpio GPIODV_24 GPIO_ACTIVE_HIGH>;
+		led_gpio = <&gpio GPIOAO_9 GPIO_ACTIVE_HIGH>;
 		led_active_low = <1>;
 	};
 
@@ -447,7 +447,7 @@
 		alarm_reg_addr = <0xc81000a8>;
 		timer_e_addr = <0xc1109988>;
 		init_date = "2015/01/01";
-		status = "okay";
+		status = "disabled";
 	};
 
 	wifi{
@@ -659,7 +659,7 @@
 		system:system
 		{
 			pname = "system";
-			size = <0x0 0x40000000>;
+			size = <0x0 0x80000000>;
 			mask = <1>;
 		};
 		cache:cache
@@ -741,6 +741,7 @@
 		keysn_8:key_8{
 			key-name = "widevinekeybox";
 			key-device = "secure";
+			key-type = "sha1";
 			key-permit = "read","write","del";
 		};
 		keysn_9:key_9{
@@ -787,9 +788,16 @@
 		gpio_i2c_en = <0x1>;
 		#address-cells = <2>;
 		#size-cells = <2>;
+		pwr-ctrl = <&pwr_ctrl>;
 		ranges;
 	};
 
+	pwr_ctrl: pwr_ctrl{
+		pwr_5v_on = <&gpio       GPIODV_7       GPIO_ACTIVE_HIGH>;
+		PWR_STBY_PWREN = <&gpio       GPIODV_4       GPIO_ACTIVE_HIGH>;
+		vol_switch = <&gpio GPIOH_5 GPIO_ACTIVE_HIGH>;
+	};
+
 	aocec: aocec{
 		compatible = "amlogic, amlogic-aocec";
 		device_name = "aocec";
@@ -1051,7 +1059,7 @@
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_codec_T9015";
 		reg = <0x0 0xc8832000 0x0 0x14>;
-		status = "okay";
+		status = "disable";
 	};
 	aml_m8_snd {
 		compatible = "aml, aml_snd_m8";
@@ -1062,7 +1070,7 @@
 				"Ext Spk","LOUTL",
 				"Ext Spk","LOUTR";
 
-		mute_gpio-gpios = <&gpio GPIOH_5 0>;
+		mute_gpio-gpios = <&gpio GPIODV_22 0>;
 		mute_inv;
 		hp_disable;
 		hp_paraments = <800 300 0 5 1>;
@@ -1081,7 +1089,7 @@
 			sound-dai = <&pcm_dai>;
 		};
 		codec0: codec0 {
-			sound-dai = <&amlogic_codec>;
+			sound-dai = <&es8323>;
 		};
 		codec1: codec1 {
 			sound-dai = <&spdif_codec>;
@@ -1257,6 +1265,39 @@
 	};
 };
 
+&i2c_a {
+  status = "okay";
+	es8323: es8323@10 {
+		#sound-dai-cells = <0>;
+		compatible = "es, es8323";
+		reg = <0x10>;
+	};
+};
+
+&i2c_b {
+  status = "okay";
+    rtc_hym8563{
+	compatible = "amlogic, rtc_hym8563";                      /** for driver probe, must have  */
+	dev_name = "rtc_hym8563";
+	status = "okay";         
+	reg = <0x51>;                                     /** device i2c address, must have        */
+	interrupts = <0 64 1>;
+	gpio-rtc-irq = <&gpio       GPIODV_3       GPIO_ACTIVE_HIGH>;
+        };
+
+	minix_mcu{
+		compatible = "amlogic,minix_mcu";
+		status = "okay";
+		reg = <0x15>;
+	};
+
+	minix_mcu_isp{
+		compatible = "amlogic,minix_mcu_isp";
+		reg = <0x35>;
+	};
+};
+
+
 &pinmux {
 	audio_pins:audio_pin{
 		amlogic,setmask=<6 0x7800000>;
